packages/@aws-cdk/aws-batch/lib/compute-environment.ts

```diff
@@ -0,0 +1,484 @@
+   *
+   * @default - Role using the 'service-role/AWSBatchServiceRole' policy
+   */
+  readonly serviceRole?: iam.IRole,
+
+  /**
+   * Determines if AWS should manage the allocation of compute resources for processing jobs.
+   * If set to false, then you are in charge of providing the compute resource details.
+   *
+   * @default true
+   */
+  readonly managed?: boolean;
+}
+
+/**
+ * Properties of a compute environment
+ */
+export interface IComputeEnvironment extends IResource {
+  /**
+   * The ARN of this compute environment
+   *
+   * @attribute
+   */
+  readonly computeEnvironmentArn: string;
+
+  /**
+   * The name of this compute environment
+   *
+   * @attribute
+   */
+  readonly computeEnvironmentName: string;
+}
+
+/**
+ * Batch Compute Environment
+ *
+ * Defines a batch compute environment to run batch jobs on.
+ */
+export class ComputeEnvironment extends Resource implements IComputeEnvironment {
+  /**
+   * Fetches an existing batch compute environment by its amazon resource name.
+   *
+   * @param scope
+   * @param id
+   * @param computeEnvironmentArn
+   */
+  public static fromComputeEnvironmentArn(scope: Construct, id: string, computeEnvironmentArn: string): IComputeEnvironment {
+    const stack = Stack.of(scope);
+    const computeEnvironmentName = stack.parseArn(computeEnvironmentArn).resource;
+
+    class Import extends Resource implements IComputeEnvironment {
+      public readonly computeEnvironmentArn = computeEnvironmentArn;
+      public readonly computeEnvironmentName = computeEnvironmentName;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The ARN of this compute environment
+   *
+   * @attribute
+   */
+  public readonly computeEnvironmentArn: string;
+
+  /**
+   * The name of this compute environment
+   *
+   * @attribute
+   */
+  public readonly computeEnvironmentName: string;
+
+  constructor(scope: Construct, id: string, props: ComputeEnvironmentProps = { enabled: true, managed: true }) {
+    super(scope, id, {
+      physicalName: props.computeEnvironmentName,
+    });
+
+    this.validateProps(props);
+
+    const spotFleetRole = this.getSpotFleetRole(props);
+    let computeResources: CfnComputeEnvironment.ComputeResourcesProperty | undefined;
+
+    // Only allow compute resources to be set when using UNMANAGED type
+    if (props.computeResources && !this.isManagedByAWS(props)) {
+      computeResources = {
+        allocationStrategy: props.allocationStrategy || AllocationStrategy.BEST_FIT,
+        bidPercentage: props.computeResources.bidPercentage,
+        desiredvCpus: props.computeResources.desiredvCpus,
+        ec2KeyPair: props.computeResources.ec2KeyPair,
+        imageId: props.computeResources.image && props.computeResources.image.getImage(this).imageId,
+        instanceRole: props.computeResources.instanceRole
+        ? props.computeResources.instanceRole.roleArn
+        : new iam.LazyRole(this, 'Resource-Instance-Role', {
```

Comment:
Any reason you used `LazyRole` here instead of just `Role`?

Reply:
I see. I was using it thinking it would simplify things a bit but I see now that regardless, the role is going to be used in the event that its declared here. I will switch it out.